server/server.go

```diff
@@ -227,9 +227,16 @@ type (
 		PathParameters map[string]string
 		// RequestPath the string representing requested resources i.e. /api/v1/organizations/:orgID/...
 		RequestPath string
+		// LoggingMetadata
+		LoggingMetadata LoggingMetadata
 	}
 
-	requestDetailsKey struct{}
+	LoggingMetadata struct {
+		Logger   *zap.SugaredLogger
+		Metadata []any
+	}
+
+	RequestDetailsKey struct{}
```

Comment:
I'd rather keep this struct private and relay on AddRequestDetailsToCtx (that's public for test  purposes only) and ExtractRequestDetailsFromContext. 
If we make the requestKey public - someone may accidently override context with other set of values...

Reply:
I managed to keep `requestDetailsKey` private, I need to access to this value from different scenarios e.g when running on an HTTP request (using context.Context) or when running on a workflow (using workflow.Context which is a Temporal's implementation of context.Context), what I did to keep the struct private was to add a new interface that I can use to be able to receive either those two context type